src/Tuura/Fantasi/HubRewrite.hs

```diff
@@ -0,0 +1,204 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Tuura.Fantasi.HubRewrite where
+
+import Pangraph
+import qualified Pangraph.FGL as FGL
+import qualified Pangraph.GraphML.Parser as GraphML
+
+import qualified Data.Graph.Inductive.Graph as FGL
+import qualified Data.Graph.Inductive.PatriciaTree as FGLTree
+
+import Data.ByteString.Char8 (pack)
+import Data.ByteString(ByteString)
+import qualified Data.ByteString as BS
+
+import Data.List(maximumBy)
+
+import Data.Maybe(fromJust)
+
+import Data.Set(Set)
+import qualified Data.Set as Set
+
+import Control.Applicative((<*>))
+
+
+-- | Used for a 'Set' implementation for 'Eq' and 'Ord' instances on endpoints only.
+newtype OrderEdgeByEnds = OrderEdgeByEnds Edge deriving (Show)
+
+-- Eq on endpoints.
+instance Eq OrderEdgeByEnds where
+    (==) (OrderEdgeByEnds a) (OrderEdgeByEnds b) = edgeEndpoints a == edgeEndpoints b 
+
+-- Ord on endpoints.
+instance Ord OrderEdgeByEnds where
+    compare (OrderEdgeByEnds a) (OrderEdgeByEnds b) = compare (edgeEndpoints a) (edgeEndpoints b)
+
+driveN1 :: IO ()
```

Comment:
What is the purpose of this function? It seems like the testsuite would be a better place for it.